### ISSUE TITLE
SimCLR Loss: False Output (Possibly with Correction)

### ISSUE THREAD

**Issue by vaydemir3** [HobbitLong/SupContrast#106]
Hi,

First of all, thanks a lot for sharing your code :) I was having an issue with a particular input to the SimCLR Loss.
Here is the code that produces the false output:
``` python
# passing a case where each feature vector is same
featTensor = torch.ones((8, 2, 10)) # batch size of 8 with 2 augmented views and feature dimension of 10
featTensor = nn.functional.normalize(featTensor, dim=-1) #l2 normalize the feature vectors
criterion = SupConLoss(temperature=0.07)
loss = criterion(featTensor)
```
When each feature vector is same, the loss should output log_e(2N-1), [to see why]( https://www.overleaf.com/project/624b22f212931470307f6fb4)

The above code fails to output log_e(2N-1)

The correct usage of the code seems to be dependent on setting both tempreature and base_temperature arguments of the loss function: 
``` python
import torch.nn as nn                                   
import numpy as np                                      
                                                        
from losses import SupConLoss                           
                                                        
the_answer = np.log(15)          
print("the answer we want", the_answer)                 
                                                        
featTensor = torch.ones((8, 2, 10))                     
featTensor = nn.functional.normalize(featTensor, dim=-1)
                                                        
criterion = SupConLoss(                                 
        temperature=1.0,                                
        base_temperature=1.0)                           
                                                        
loss = criterion(featTensor)
print("the answer we get", loss)   
```

**Comment by tbenst** [comment]
Thanks for this! I was trying to figure out the purpose of `base_temperature`. It does appear that the code here is different than in the original paper. Curious if anyone has a justification for the presence of base_temperature..? Wasn’t sure if there was a subsequent contrastive loss paper that introduces the idea or if the authors here found it an empirically useful hyperparameter.

**Comment by adv010** [comment]
@tbenst I was also curious about the base_temperature parameter. Did you get any insights / applications for the same?
